plugin/src/main/java/org/eobjects/build/DotnetNugetPushMojo.java

```diff
@@ -32,8 +32,13 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         for (File subDirectory : helper.getProjectDirectories()) {
             try {
                 final File nugetPackage = helper.getNugetPackage(subDirectory);
+                final String repositoryKey = helper.getRepositoryKey();
                 final String nugetPackagePath = nugetPackage.getCanonicalPath();
-                helper.executeCommand(subDirectory, "dotnet", "nuget", "push", nugetPackagePath, "-s", repository);
+                if(repositoryKey == null || repositoryKey.isEmpty())
+                {
+                	helper.executeCommand(subDirectory, "dotnet", "nuget", "push", nugetPackagePath, "-s", repository);
+                }
+                else helper.executeCommand(subDirectory, "dotnet", "nuget", "push", nugetPackagePath, "-k", repositoryKey , "-s", repository);
```

Comment:
I suggest that we build up the command line more dynamically here, or else we'll just end up with more and more `if` statements. So I'd rewrite the above to something like:

```
final List<String> cmd = new ArrayList<>(Arrays.asList("dotnet", "nuget", "push"));
cmd.add(nugetPackagePath);
cmd.add("-s");
cmd.add(repository);
if (repositoryKey != null && !repositoryKey.isEmpty) {
  cmd.add("-k");
  cmd.add(repositoryKey);
}
helper.executeCommand(subDirectory, cmd);
```

Also, please be aware that since this is written in Java, we should follow java coding guidelines/norms. So no separate line just for the starting curly brace. And a space after `if`, before the `(` :-)